### DISCUSSION TITLE
基于文件来管理部署多个bfe实例， 文件的修改之后的同步  怎样设计呢 保证及时

### DISCUSSION THREAD

**Discussion by wanghonglei5181** [bfenetworks/bfe#814] [Q&A (问题和回答)] [answered]
配置生成，管理，下发的设计方案能简要描述一下么？

**Comment by liuqing6767** [comment, marked as answer]
感谢提问。

数据平面专注于流量接入和转发，控制相关逻辑的配置更新当前的设计是通过调用reload APi读取本地配置文件完成加载逻辑。

在小规模低频度的场景下，配置变更通过手工维护即可满足需求。大规模自动化运维场景就需要控制面的配合：
- 感知配置变更，生成最新的配置文件
- 触发数据平面加载

正如你的提问提到的，里面需要考虑如下几个问题：
- 配置管理：API Server 会提供全套的 OpenAPI 完成配置管理
- 配置生成的完整性：每次都有版本号，对于相互依赖的配置同时生成所有的配置文件
- 配置下发的实时性：推模式或者拉模式。这里面有很多技术选型。但是有一点是明确的，就是会和bfe同机部署一个agent
- 大规模场景下的性能：需要的话配置生成节点可以进行多级部署

开源版本给出如下最简单部署方案，在小规模场景下既能满足业务需求，又减轻运维负担
![deploy architecture](https://user-images.githubusercontent.com/9527871/132456142-46efcb10-71f4-47f8-8537-4d639caeea88.png)
